addon/index.js

```diff
@@ -172,9 +171,61 @@ function formatInstallData(install, addon) {
   return formatted;
 }
 
+function isIdeatownAddonID(id) {
+  return id in store.availableExperiments;
+}
+
+function syncAddonInstallation(addonID) {
+  const experiment = store.availableExperiments[addonID];
+  const method = (addonID in store.installedAddons) ? 'put' : 'delete';
+  // HACK: Use the same "done" handler for 2xx & 4xx responses -
+  // 200 = PUT success, 410 = DELETE success, 404 = DELETE redundant
+  const done = (res) => [addonID, method, res.status];
+  return requestAPI({
+    method: method,
+    url: experiment.installations_url + store.clientUUID
+  }).then(done, done);
+}
+
+function requestAPI(opts) {
+  const headers = {
+    'Accept': 'application/json',
+    'Cookie': ''
+  };
+
+  const hostname = URL(prefs.BASE_URL).hostname; // eslint-disable-line new-cap
+  const cookieEnumerator = cookieManager2.getCookiesFromHost(hostname);
+  while (cookieEnumerator.hasMoreElements()) {
+    const c = cookieEnumerator.getNext().QueryInterface(Ci.nsICookie); // eslint-disable-line new-cap
+    headers.Cookie += c.name + '=' + c.value + ';';
+    if (c.name === 'csrftoken') {
+      headers['X-CSRFToken'] = c.value;
+    }
+  }
+
+  return new Promise((resolve, reject) => {
+    request({
+      url: opts.url,
+      headers: Object.assign(headers, opts.headers || {}),
+      contentType: 'application/json',
+      onComplete: res => (res.status < 400) ? resolve(res) : reject(res)
+    })[opts.method || 'get']();
+  });
+}
+
+// source: http://jsfiddle.net/briguy37/2mvfd/
```

Comment:
Figured the full node-uuid module would be overkill, since we only ever need a vaguely random UUID. This is actually what the README links to from "Want something smaller? Check this crazy shit out!" :)